Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -2192,8 +2192,18 @@ public UUID addDelayedTriggeredAbility(DelayedTriggeredAbility delayedAbility, A
 
     @Override
     public UUID fireReflexiveTriggeredAbility(ReflexiveTriggeredAbility reflexiveAbility, Ability source) {
+        return fireReflexiveTriggeredAbility(reflexiveAbility, source, false);
+    }
+    
+    @Override
+    public UUID fireReflexiveTriggeredAbility(ReflexiveTriggeredAbility reflexiveAbility, Ability source, boolean fireAsSimultaneousEvent) {
         UUID uuid = this.addDelayedTriggeredAbility(reflexiveAbility, source);
-        this.fireEvent(GameEvent.getEvent(GameEvent.EventType.OPTION_USED, source.getOriginalId(), source, source.getControllerId()));
+        GameEvent event = GameEvent.getEvent(GameEvent.EventType.OPTION_USED, source.getOriginalId(), source, source.getControllerId());
+        if (fireAsSimultaneousEvent) {
+            this.getState().addSimultaneousEvent(event, this);
```

Comment:
> 603.12a Normally, if the trigger event or events occur multiple times during the resolution of the spell or ability that created it, the reflexive triggered ability will trigger once for each of those times. However, if a resolving spell or ability includes a choice to pay a cost “any number of times” and creates a triggered ability that triggers “when [a player] pays [that cost] one or more times,” paying that cost one or more times causes the reflexive triggered ability to trigger only once.

OPTION_USED need to execute it immediately.

If your use case generates multiple reflexive triggers then must execute only once (see rules above). So that’s looks like a game engine bug and simultaneous event must be used in all reflexive triggers.

Reply:
Must be a workable use case:

If an amount of damage would be dealt to Magma Pummeler greater than the number of +1/+1 counters on it, all of that damage is prevented, and all of those counters are removed. In most cases, this will result in Magma Pummeler having 0 toughness, so it will be put into its owner's graveyard. Its reflexive triggered ability still triggers and it will still deal that much damage to any target.
(2021-11-19)

Reply:
> Smells bed. Reflexive trigger fires immediately due rules. Why you need to change it to simulate normal trigger?

It's still a reflexive trigger and it is fired at the appropriate time. Simultaneous event used for batching. There is no batch for replacement events, so this is necessary for Magma Pummeler to function correctly (it must only fire after all simultaneous replacements have been processed).

(Whether all reflexive triggers should fire as simultaneous events is a separate question but a larger scope than this PR)

Reply:
Anyway magma die use case must be covered by test (see above)

Reply:
Sure, added 2 tests, dealing more than the amount of removed counters was never the issue there.

And xeno's answer is on point, the reflexive is created with first replacement effect, but there might be more simultaneous ones waiting, and we can't fire the event triggering the reflexive effect until all are done. Using the simultaneousEvent queue does delay the firing trigger until after all replacement are done.

A full rework of prevention effect could in theory wait for them all before firing resulting reflexive trigger, but that's huge amount of work, and probably not worth it for the single Pummeler card. The ad hoc solution here works.